content/technotes/2020-07-07-github-pat-credentials.md

```diff
@@ -0,0 +1,99 @@
+---
+slug: github-pat
+title: 'A better way to manage your Github personal access tokens'
+date: '2020-07-07'
+author:
+  - Jeroen Ooms
+tags:
+  - R
+  - package
+  - gert
+  - credentials
+  - git
+  - github
+---
+
+We have been working hard behind the scenes on the upcoming release of our new git package named [gert](https://docs.ropensci.org/gert), a joint effort from rOpenSci and the Tidyverse team. One of the main features of gert is the out-of-the-box authentication mechanism, which is provided via the new [credentials](https://docs.ropensci.org/credentials/articles/intro.html) package.
+
+Among other things, the credentials package makes it possible to save and load https authentication details from the [git credential store](https://git-scm.com/docs/git-credential-store), which is part of the official command-line git. Thereby credentials are automatically shared between command line git and the gert package, while safely stored by your operating system preferred password manager.
+
+In this post we show how you can take this one step further, and use the credentials package to save your `GITHUB_PAT` in the git credential store. This way you can authenticate with the GitHub API using the same token that is used for HTTPS remotes in git and gert. This is convenient for users, and also provides package authors with a mechanism to prompt the user for credentials, without having to take responsibility for managing tokens.
+
+
+## What is a Personal Access Token
+
+GitHub allows you to generate [Personal Access Tokens](https://github.com/settings/tokens), which you can use instead of your password when authenticating over HTTPS, both for git remotes and the GitHub API. Major advantages are:
+
+ - If you have enabled 2FA, you need a PAT to authenticate programmatically.
+ - A PAT can easily be revoked or updated. You can even automate this via the Github API.
```

Comment:
Updated? I don't think that's true.My impression is that if you need to get the actual PAT or update its scopes, you have to start over. But the ability to revoke individual PATs seems like a really important point.

Reply:
Hmm I thought the Github API would let you generate and revoke PATs? 🤔